FileProcessor/src/main/java/elements/Rename.java

```diff
@@ -22,7 +25,20 @@ public Rename() {
 
     @Override
     public void process() {
-        throw new UnsupportedOperationException("Not supported yet.");
+        ArrayList<Entry> sufName = this.getInputEntries();// creates array that is the input entries 
+        ArrayList<Entry> output = new ArrayList(); //creates a output array 
+        for (Entry e : sufName) { //created a for loop that accesses the addresses within the array to create a string, str
+            String str = e.getAddress();
+            Entry renamedEntry = e.clone(); // clone the input entries string 
+            int index = str.lastIndexOf(".");// search through the string to find the last instance of '.'
```

Comment:
You're missing support for folders. This will always work on a LocFile or RemFile, but a LocFolder or a RemFolder might not have a '.' anywhere in its name.